src/main/scala/is/hail/expr/types/TStruct.scala

```diff
@@ -90,6 +90,13 @@ final case class TStruct(fields: IndexedSeq[Field], override val required: Boole
     case _ => false
   }
 
+  def isIsomorphicTo(other: TStruct): Boolean =
+    size == other.size && isPrefixOf(other)
```

Comment:
isn't this exactly `isOfType(other)`

Reply:
`isOfType` compares names also. There are places, such as testing for joinability, where you want to ignore the top-level names of the struct.

Since the lower-level names still matter, I'm not really happy with the name `isIsomorphicTo`. Another option would be to put a field on `TStruct` which gives the corresponding `Tuple`, and use `isOfType` on the `Tuple`s.

Reply:
ahhh, I see. This is fine, then.